mirdata/datasets/queen.py

```diff
@@ -0,0 +1,262 @@
+        filename="Queen Annotations.tar.gz",
+        url="http://isophonics.net/files/annotations/Queen%20Annotations.tar.gz",
+        checksum="fe11217d32bc222ae418425441974046",
+        destination_dir="annotations",
+    )
+}
+
+
+DOWNLOAD_INFO = """
+        Unfortunately the audio files of Queen dataset are not available
+        for download. If you have Queen dataset, place the contents into
+        a folder called Queen with the following structure:
+            > Queen/
+                > annotations/
+                > audio/
+        and copy Queen folder to {}
+"""
+
+
+class Track(core.Track):
+    """Queen track class
+
+    Args:
+        track_id (str): track id of the track
+
+    Attributes:
+        audio_path (str): track audio path
+        chords_path (str): chord annotation path
+        keys_path (str): key annotation path
+        sections_path (str): sections annotation path
+        title (str): title of the track
+        track_id (str): track id
+
+    Cached Properties:
+        chords (ChordData): human-labeled chord annotations
+        key (KeyData): local key annotations
+        sections (SectionData): section annotations
+    """
+
+    def __init__(
+        self,
+        track_id,
+        data_home,
+        dataset_name,
+        index,
+        metadata,
+    ):
+        super().__init__(
+            track_id,
+            data_home,
+            dataset_name,
+            index,
+            metadata,
+        )
+
+        self.chords_path = self.get_path("chords")
+        self.keys_path = self.get_path("keys")
+        self.sections_path = self.get_path("sections")
+        self.audio_path = self.get_path("audio")
+
+        self.title = os.path.basename(self.get_path("sections")).split(".")[0]
+
+    @core.cached_property
+    def chords(self) -> Optional[annotations.ChordData]:
+        return load_chords(self.chords_path)
+
+    @core.cached_property
+    def key(self) -> Optional[annotations.KeyData]:
+        return load_key(self.keys_path)
+
+    @core.cached_property
+    def sections(self) -> Optional[annotations.SectionData]:
+        return load_sections(self.sections_path)
+
+    @property
+    def audio(self) -> Optional[Tuple[np.ndarray, float]]:
+        """The track's audio
+
+        Returns:
+            * np.ndarray - audio signal
+            * float - sample rate
+
+        """
+        return load_audio(self.audio_path)
+
+    def to_jams(self):
+        """the track's data in jams format
+
+        Returns:
+            jams.JAMS: return track data in jam format
+
+        """
+        return jams_utils.jams_converter(
+            audio_path=self.audio_path,
+            section_data=[(self.sections, None)],
+            chord_data=[(self.chords, None)],
+            key_data=[(self.key, None)],
+            metadata={"artist": "The Queen", "title": self.title},
+        )
+
+
+def load_audio(fhandle: TextIO) -> Tuple[np.ndarray, float]:
```

Comment:
Missing `io` decorator

Reply:
done

Reply:
marked as done, but the decorator isn't there

Reply:
done now!